src/main/webapp/app/exercises/modeling/manage/modeling-exercise-update.component.ts

```diff
@@ -211,27 +225,36 @@ export class ModelingExerciseUpdateComponent implements OnInit {
         );
     }
 
-    /**
-     * Revert to the previous state, equivalent with pressing the back button on your browser
-     * Returns to the detail page if there is no previous state
-     * Returns to the overview page if there is no previous state and we created a new exercise
-     * Returns to the exercise groups page if we are in exam mode
-     */
     previousState() {
         navigateBackFromExerciseUpdate(this.router, this.modelingExercise);
     }
 
     private subscribeToSaveResponse(result: Observable<HttpResponse<ModelingExercise>>): void {
         result.subscribe(
-            () => this.onSaveSuccess(),
+            (exercise: HttpResponse<ModelingExercise>) => this.onSaveSuccess(exercise.body!.id!),
             () => this.onSaveError(),
         );
     }
 
-    private onSaveSuccess(): void {
+    private onSaveSuccess(exerciseId: number): void {
         this.eventManager.broadcast({ name: 'modelingExerciseListModification', content: 'OK' });
         this.isSaving = false;
-        this.previousState();
+
+        switch (this.editType) {
+            case EditType.CREATE:
+                // Assigning id received from the backend.
+                // Required for navigation to the example submission dashboard in case of the exercise CREATE.
+                this.modelingExercise.id = exerciseId;
+
+                setTimeout(() => {
```

Comment:
Is there a particular reason why you're using a timeout? Why not just execute the function without a 1 second delay?

Reply:
yes there is a reason @ivanchimeno 
I delayed the redirection with 1 second in order for the user to be able to see the alert which shows that exercise was created successfully.

Reply:
just a thought: what if it takes longer than 1 second for some reason...you wait and still are not able to see the notification. Would it be too complex to link the navigation to the actual appearance of the notification?